package/kedro_viz/api/rest/router.py

```diff
@@ -38,53 +38,48 @@ async def main():
-
-    if isinstance(node, TaskNode):
-        return TaskNodeMetadata(node)
-
-    if isinstance(node, DataNode):
-        return DataNodeMetadata(node)
-
-    if isinstance(node, TranscodedDataNode):
-        return TranscodedDataNodeMetadata(node)
-
-    return ParametersNodeMetadata(node)
+    return get_node_metadata_response(node_id)
 
 
 @router.get(
     "/pipelines/{registered_pipeline_id}",
     response_model=GraphAPIResponse,
 )
 async def get_single_pipeline_data(registered_pipeline_id: str):
-    if not data_access_manager.registered_pipelines.has_pipeline(
-        registered_pipeline_id
-    ):
-        return JSONResponse(status_code=404, content={"message": "Invalid pipeline ID"})
-
-    modular_pipelines_tree = (
-        data_access_manager.create_modular_pipelines_tree_for_registered_pipeline(
-            registered_pipeline_id
+    return get_selected_pipeline_response(registered_pipeline_id)
+
+
+@router.post("/deploy")
+async def deploy_kedro_viz(input_values: S3DeployerConfiguration):
+    try:
+        deployer = S3Deployer(input_values.region, input_values.bucket_name)
+        url = deployer.deploy_and_get_url()
+        response = {"message": "Website deployed on S3", "url": url}
+        return JSONResponse(status_code=200, content=response)
+    except PermissionError as exc:  # pragma: no cover
+        logger.exception("Permission error in deploying Kedro Viz : %s ", exc)
+        return JSONResponse(
+            status_code=401, content={"message": "Please provide valid credentials"}
+        )
+    except Exception as exc:  # pragma: no cover
+        logger.exception("Deploying Kedro Viz failed: %s ", exc)
+        return JSONResponse(
+            status_code=500, content={"message": "Failed to deploy Kedro Viz"}
         )
-    )
 
-    return GraphAPIResponse(
-        nodes=data_access_manager.get_nodes_for_registered_pipeline(  # type: ignore
-            registered_pipeline_id
-        ),
-        edges=data_access_manager.get_edges_for_registered_pipeline(  # type: ignore
-            registered_pipeline_id
-        ),
-        tags=data_access_manager.tags.as_list(),
-        layers=data_access_manager.get_sorted_layers_for_registered_pipeline(
-            registered_pipeline_id
-        ),
-        pipelines=data_access_manager.registered_pipelines.as_list(),
-        selected_pipeline=registered_pipeline_id,
-        modular_pipelines=modular_pipelines_tree,  # type: ignore
-    )
+
+@router.get(
+    "/package-compatibilities",
+    response_model=PackageCompatibilityAPIResponse,
+)
+async def get_package_compatibilities():
+    try:
+        return get_package_compatibilities_response()
+    except Exception as exc:
+        logger.exception(
+            "An exception occured while getting package compatibility info : %s", exc
+        )
+        return JSONResponse(
+            status_code=500,
+            content={"message": "Failed to get package compatibility info"},
+        )
```

Comment:
When does this endpoint get called?

Reply:
sure, so in the FE when the user clicks on `Deploy Kedro Viz` button -- we first make a call to this endpoint /api/package-compatibilities and get the fsspec version. If the version is compatible with Shareable viz, then they will get a form to enter thier bucket-name and region. If not, then they will get a message asking them to enable Shareable viz; they need to be on a particular fsspec version. See example from the screenshot
![Uploading Screenshot 2023-09-15 at 12.47.25.png…]()